src/scenarios/utils.py

```diff
@@ -1,5 +1,30 @@
+from ipaddress import IPv4Address, IPv6Address, ip_address
+from kubernetes import client, config
+
+
 def ensure_miner(node):
     wallets = node.listwallets()
     if "miner" not in wallets:
         node.createwallet("miner", descriptors=True)
     return node.get_wallet_rpc("miner")
+
+
+def get_service_ip(service_name: str, namespace: str = "warnet") -> (IPv4Address | IPv6Address,
+                                                                     IPv4Address | IPv6Address):
+    """Given a service name and namespace, returns the service's external ip and internal ip"""
+    # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Endpoints.md
+    config.load_incluster_config()
+    v1 = client.CoreV1Api()
+    service = v1.read_namespaced_service(name=service_name, namespace=namespace)
```

Comment:
What exactly are we (planning on) using the external ip address for?

Reply:
I want to be able to get the bare ip address that belongs to the service, sometimes called the `cluster ip` address. Maybe a separate function would be a nicer way to do this?

For context, and to show you what my mental model is on this issue, here's some of what I know about these ip address:

`external` should be the ip address of the service. So, when we do `kubectl get all` and we look down at the services listed, the ip addresses listed next to each service should match the `external` address. These addresses are often called the `cluster ip` addresses, I believe.

A service provides a permanent ip (aka "cluster" ip) address. Pods have their own ip address too, and the pods "go to work" behind the cluster.

Reply:
Gotcha thanks, that's helpful. Also, because I am still a k8s noob I don't know what some of these things do...